File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTReportDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTModuleDef } from "./TTModuleDef";
import { TTPermissionDef } from "./TTPermissionDef";

export class TTReportDef {
    public ModuleDefID: Guid;
    public ModuleDef: TTModuleDef;
    public ID: Guid;
    public ReportDefID: Guid;
    public ReportNO: string;
    public CodeName: string;
    public Name: string;
    public Caption: string;
    public Description: string;
    public ReportXML: string;
    public ExcelTemplate: Object;
    public Methods: string;
    public PermissionDefID: Guid;
    public PermissionDef: TTPermissionDef;
    public ApplicationName: string;
    public ToString(): string {
        return null;
    }
}

File: AtlasModel/Enterprise/TTReportDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTReportDef
    {
        public Guid ReportDefId { get; set; }
        public string Name { get; set; }
        public string Caption { get; set; }
        public string ReportXml { get; set; }
        public Guid ModuleDefId { get; set; }
        public string Methods { get; set; }
        public string CheckOutId { get; set; }
        public short CheckOutStatus { get; set; }
        public Guid? PermissionDefId { get; set; }
        public string ReportNo { get; set; }
        public object ExcelTemplate { get; set; }
        public string Description { get; set; }
    }
}